#!/usr/bin/env python
PACKAGE = "dynamic_marker"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()



marker_family_enum = gen.enum([
gen.const("whycon",         int_t, 0, "whycon markers"),
gen.const("aruco_single",   int_t, 1, "aruco single marker"),
gen.const("aruco_multi",    int_t, 2,  "aruco board marker"),
gen.const("pitag",          int_t, 3,  "pitag marker"),
],"Set the marker family to be used")

gen.add("threshold_aruco_single", double_t, 0, "marker-camera distance smaller than this will use aruco single", 1.2, 0, 10.0)
gen.add("threshold_aruco_multi",  double_t, 0, "marker-camera distance smaller than this will use aruco board",  0.00, 0, 10.0)
gen.add("marker_fov",             double_t, 0, "Amount of marker in FOV",                                        0.10, 0, 1.0)
gen.add("capture_delay",          double_t, 0, "Whole camera capture process delay in milliseconds",             0.23, 0, 1.0)
gen.add("marker_id",              int_t,    0, "Marker ID",                                                      88,    0, 1024)
gen.add("marker_size",            double_t, 0, "Size of the marker in meters",                                   0.15, 0, 0.15)
gen.add("max_marker_size",        double_t, 0, "Maximum size of the marker in meters. (depends on the display screen)", 0.15, 0, 0.15)
gen.add("marker_family",          int_t,    0, "Set the marker family to be used", 0, 0, 3, edit_method=marker_family_enum)
gen.add("marker_frame_name",      str_t,    0, "Name of the marker (it is used as the name of the coordinate frame of the marker)", "land_target")

exit(gen.generate(PACKAGE, "dynamic_marker", "dynamic_param_config"))

